parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -138,30 +153,41 @@ public Metrics metrics() {
   @Override
   public long length() {
     try {
-      if (closed) {
-        return writer.getPos();
-      } else {
-        return writer.getPos() + (writeStore.isColumnFlushNeeded() ? writeStore.getBufferedSize() : 0);
+      long length = 0L;
+
+      if (writer != null) {
+        length += writer.getPos();
       }
+
+      if (!closed && recordCount > 0) {
+        // recordCount > 0 when there are records in the write store that have not been flushed to the Parquet file
+        length += writeStore.getBufferedSize();
```

Comment:
I know this isn't realated to this pr, but what is `getBufferedSize()` when record count == 0 or the writer is closed? Just seems odd that we can't always add this on. NBD just wondering